src/__tests__/extensions/sessionrecording.js

```diff
@@ -206,43 +214,69 @@ describe('SessionRecording', () => {
             expect(given.sessionRecording.captureStarted).toEqual(false)
         })
 
-        it('sends a full snapshot if there is a new session id and its not type 2 or 4', () => {
-            sessionIdGenerator.mockReturnValue({
-                isNewSessionId: true,
-                sessionId: 'new-sid',
-            })
+        describe('session and window ids', () => {
+            beforeEach(() => {
+                given.sessionRecording.sessionId = 'old-session-id'
+                given.sessionRecording.windowId = 'old-window-id'
 
-            given.sessionRecording.startRecordingIfEnabled()
-            given.sessionRecording.submitRecordings()
+                given.sessionRecording.startRecordingIfEnabled()
+                given.sessionRecording.submitRecordings()
+            })
 
-            _emit({ event: 123, type: 3 })
-            expect(window.rrweb.record.takeFullSnapshot).toHaveBeenCalled()
-        })
+            it('sends a full snapshot if there is a new session id and the event is not type FullSnapshot or Meta', () => {
+                given.posthog._sessionIdManager.getSessionAndWindowId.mockReturnValue({
+                    sessionId: 'new-session-id',
+                    windowId: 'old-window-id',
+                })
 
-        it('does not send a full snapshot if there is a new session id and its type 2 or 4', () => {
-            sessionIdGenerator.mockReturnValue({
-                isNewSessionId: true,
-                sessionId: 'new-sid',
+                _emit({ event: 123, type: INCREMENTAL_SNAPSHOT_EVENT_TYPE })
+                expect(window.rrweb.record.takeFullSnapshot).toHaveBeenCalled()
             })
 
-            given.sessionRecording.startRecordingIfEnabled()
-            given.sessionRecording.submitRecordings()
+            it('sends a full snapshot if there is a new window id and the event is not type FullSnapshot or Meta', () => {
+                given.posthog._sessionIdManager.getSessionAndWindowId.mockReturnValue({
+                    sessionId: 'old-session-id',
+                    windowId: 'new-window-id',
+                })
 
-            _emit({ event: 123, type: 4 })
-            expect(window.rrweb.record.takeFullSnapshot).not.toHaveBeenCalled()
-        })
+                _emit({ event: 123, type: INCREMENTAL_SNAPSHOT_EVENT_TYPE })
+                expect(window.rrweb.record.takeFullSnapshot).toHaveBeenCalled()
+            })
 
-        it('does not send a full snapshot if there is not a new session', () => {
-            sessionIdGenerator.mockReturnValue({
-                isNewSessionId: false,
-                sessionId: 'sid',
+            it('does not send a full snapshot if there is a new session/window id and the event is type FullSnapshot or Meta', () => {
```

Comment:
The meta event is sent right before the full snapshot. It contains information about the window size etc. The reason we don't send a full snapshot in this case is because it always happens right before a full snapshot, so rr-web is already in the process of taking a snapshot.

Reply:
Nice, thanks for the clarification :)